bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/ReplicationWorker.java

```diff
@@ -186,9 +187,12 @@ private static void waitBackOffTime() {
 
     private void waitTillTargetBookieIsWritable() {
         LOG.info("Waiting for target bookie {} to be back in read/write mode", targetBookie);
-        while (admin.getReadOnlyBookies().contains(targetBookie)) {
+        while (workerRunning && admin.getReadOnlyBookies().contains(targetBookie)) {
+            isInReadOnlyMode = true;
             waitBackOffTime();
```

Comment:
We can check from ZK that it really went to RW mode instead of blind wait right?